### PR TITLE
LegendItem: add `Parent` property to pair items with plottables

### PR DESCRIPTION
**Pull request by BambOoxX**
Add legend viewer functionnality to `ScottPlot.WinForms` as in SP4. See #3842 and #4147

@swharden there are some points I'd need help with :

- I could not reproduce the "graying" effect that was present in SP4 to show that a plot is hidden after clicking on the `LegendItem`. There is probably some additional SK painting to perform, but I don't really know how to do this properly
- There is a strange behavior when trying to change marker sizes, the behavior feels unnatural (marker size vs marker line width) 

There are probably other things to improve. Let me know.

### CODE REVIEW

**Comment by swharden** [comment]
Hey @BambOoxX, thanks so much for your work on this!

I know we can work toward implementing a detachable legend viewer. However, my recommendation is to do this in two pull requests to keep things simple and moving fast.

Can we adapt this PR to not modify anything in the WinForms namespace? Adding that pop-out control is something we should probably put in the demo app (not ship with the ScottPlot package) and we can discuss that in a separate PR.

Similarly, can we remove `IsDetached` and `IsToggled` from `Legend`? I'd like to do whatever we can to keep the core ScottPlot project GUI-free (this property doesn't make sense for console apps), so perhaps we can solve that problem later in that second pull request. I'm thinking a `Dictionary<Legenditem, bool>` or maybe a custom class could keep track of GUI-related modifications to legend items, but again, more discussion for that second PR...

If this PR becomes GUI-free, I can merge it today and it will definitely be released in the next release!

File: src/ScottPlot5/ScottPlot5/Primitives/Legend.cs [hunk 1]
```diff
@@ -123,7 +125,7 @@ public LegendItem[] GetItems()
         if (DisplayPlottableLegendItems)
         {
             var plottableLegendItems = Plot.PlottableList
-                        .Where(item => item.IsVisible)
+                        // .Where(item => item.IsVisible)
```
Comment (swharden) [inline src/ScottPlot5/ScottPlot5/Primitives/Legend.cs:128]: Is this intentional?

**Comment by BambOoxX** [comment]
I'll see to modify the implementation per your observations tomorrow.

**Comment by BambOoxX** [comment]
All changes you be limited to plottables right now. I reverted all changes related to the GUI.

**Comment by swharden** [comment]
Thanks @BambOoxX! I'll refine this a bit, merge, then release a new package 🚀

**Comment by swharden** [comment]
Sure thing! 

I don't have it meticulously planned out in my head, but is it something that could be added to the demo app without requiring code modification of the core library?

**Comment by swharden** [comment]
Sounds good! No rush, but I'm here if you hit any snags.

I don't have it all figured out in my head - I'm just assuming it won't be too hard - but it will be interesting to see how this one plays out!

**Pull request merged**